exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerTest.java

```diff
@@ -295,6 +302,38 @@ void stringRepresentation() {
         .isEqualTo("PrometheusHttpServer{address=" + prometheusServer.getAddress() + "}");
   }
 
+  @Test
+  void defaultExecutor() {
+    assertThat(prometheusServer)
+        .extracting("executor", as(InstanceOfAssertFactories.type(ThreadPoolExecutor.class)))
+        .satisfies(
+            executor -> {
+              assertThat(executor.getCorePoolSize()).isEqualTo(5);
+            });
+  }
+
+  @Test
+  void customExecutor() throws IOException {
+    ScheduledExecutorService scheduledExecutor = Executors.newScheduledThreadPool(10);
+    int port;
+    try (ServerSocket socket2 = new ServerSocket(0)) {
+      port = socket2.getLocalPort();
+    }
+    PrometheusHttpServer server =
+        PrometheusHttpServer.builder()
+            .setHost("localhost")
+            .setPort(port)
+            .setExecutor(scheduledExecutor)
+            .build();
+    assertThat(server)
+        .extracting(
+            "executor", as(InstanceOfAssertFactories.type(ScheduledThreadPoolExecutor.class)))
+        .satisfies(
+            executor -> {
+              assertThat(executor.getCorePoolSize()).isEqualTo(10);
```

Comment:
Nit: you can use
```suggestion
              assertThat(executor).isSameAs(scheduledExecutor);
```
for a stronger assertion